src/simd.jl

```diff
@@ -0,0 +1,576 @@
+    D = SVec{1, T}(get_data(S))
+    DD = D * D
+    sDD = sum(DD)
+    return sqrt(sDD)
+end
+
+@inline function norm{T <: SIMDTypes}(S::Tensor{4, 2, T})
+    D = SVec{16, T}(get_data(S))
+    DD = D * D
+    sDD = sum(DD)
+    return sqrt(sDD)
+end
+
+@inline function norm{T <: SIMDTypes}(S::Tensor{4, 3, T})
+    @inbounds begin
+        D = get_data(S)
+        D80 = SVec{80, T}((D[1],  D[2],  D[3],  D[4],  D[5],  D[6],  D[7],  D[8],  D[9],
+                           D[10], D[11], D[12], D[13], D[14], D[15], D[16], D[17], D[18],
+                           D[19], D[20], D[21], D[22], D[23], D[24], D[25], D[26], D[27],
+                           D[28], D[29], D[30], D[31], D[32], D[33], D[34], D[35], D[36],
+                           D[37], D[38], D[39], D[40], D[41], D[42], D[43], D[44], D[45],
+                           D[46], D[47], D[48], D[49], D[50], D[51], D[52], D[53], D[54],
+                           D[55], D[56], D[57], D[58], D[59], D[60], D[61], D[62], D[63],
+                           D[64], D[65], D[66], D[67], D[68], D[69], D[70], D[71], D[72],
+                           D[73], D[74], D[75], D[76], D[77], D[78], D[79], D[80]))
+        D80D80 = D80 * D80
+        v = sum(D80D80)
+        v += D[81] * D[81]
+        return sqrt(v)
+    end
+end
+
+# rely on fast dcontract
+@inline norm{dim, T <: SIMDTypes}(S::SymmetricTensor{2, dim, T}) = sqrt(dcontract(S, S))
+
+@inline function norm{T <: SIMDTypes}(S::SymmetricTensor{4, 1, T})
+    F = SVec{1, T}((T(1), ))
+    D = SVec{1, T}(get_data(S))
+    DD = D*D; FDD = F * DD; sFDD = sum(FDD)
+    return sqrt(sFDD)
+end
+@inline function norm{T <: SIMDTypes}(S::SymmetricTensor{4, 2, T})
+    F = SVec{9, T}((T(1), T(2), T(1), T(2), T(4), T(2), T(1), T(2), T(1)))
+    D = SVec{9, T}(get_data(S))
+    DD = D * D; FDD = F * DD; sFDD = sum(FDD)
+    return sqrt(sFDD)
+end
+@inline function norm{T <: SIMDTypes}(S::SymmetricTensor{4, 3, T})
+    F = SVec{36, T}((T(1), T(2), T(2), T(1), T(2), T(1), T(2), T(4), T(4), T(2), T(4), T(2),
+                     T(2), T(4), T(4), T(2), T(4), T(2), T(1), T(2), T(2), T(1), T(2), T(1),
+                     T(2), T(4), T(4), T(2), T(4), T(2), T(1), T(2), T(2), T(1), T(2), T(1)))
+    D = SVec{36, T}(get_data(S))
+    DD = D * D; FDD = F * DD; sFDD = sum(FDD)
+    return sqrt(sFDD)
+end
+
+# dcontract
+@inline dcontract{dim, T <: SIMDTypes}(S1::SecondOrderTensor{dim, T}, S2::SecondOrderTensor{dim, T}) = dcontract(promote(S1, S2)...)
+
+@inline function dcontract{T <: SIMDTypes}(S1::Tensor{2, 3, T}, S2::Tensor{2, 3, T})
+    D1 = SVec{9, T}(get_data(S1))
+    D2 = SVec{9, T}(get_data(S2))
+    D1D2 = D1 * D2
+    return sum(D1D2)
+end
+
+@inline function dcontract{T <: SIMDTypes}(S1::SymmetricTensor{2, 3, T}, S2::SymmetricTensor{2, 3, T})
+    D1 = SVec{6, T}(get_data(S1))
+    D2 = SVec{6, T}(get_data(S2))
+    F  = SVec{6, T}((T(1), T(2), T(2), T(1), T(2), T(1)))
+    D1D2 = D1 * D2; FD1D2 = F * D1D2
```

Comment:
Something is wrong here.

```
> @btime dcontract2($a, $b)
  3.029 μs (35 allocations: 2.03 KiB)
> 1.6692358494599173
```

It seems the call to `sum` causes allocations. Furthermore, the standard non SIMD method is 3x faster for me (when removing the sum call).